portfolio/src/main/webapp/gameoflife.js

```diff
@@ -0,0 +1,144 @@
+/* 
+CODE FOR GAME OF LIFE
+    The game of life is a "cellular automota" game, where a grid of cells which can either be "alive" or "dead".
+    Each game step, all the cells are updated based upon the amount live cells around them (either becoming alive, staying alive, or dying).
+    This code contains the logic for the game of life, and displays it in a canvas element. 
+*/
+
+// Animation specific variables
+let canvas = document.getElementById("mycanvas");
+let ctx = canvas.getContext("2d");
+let stop = true;
+let interval, startTime, now, then, elapsed;
+
+resizeCanvasToDisplaySize(canvas);
+
+// Game of Life specific variables
+const SPEED = 200; // in miliseconds (speed between updates)
+const CELL_SIZE = 10; // in pixels (side length of individual cell)
+const SEED_SPAWN_RATE = 0.1; // as a ratio (likelihood a cell is 'alive' when seeded)
+
+let rows = Math.floor(canvas.height / CELL_SIZE);
+let cols = Math.floor(canvas.width / CELL_SIZE);
+
+// Declare grid for Game of Life
+let grid = new Array(rows);
+for (let i = 0; i < grid.length; i++) {
+	grid[i] = new Array(cols);
+}
+
+// Animation "on/off" function (toggles whether game of life is running or not)
+function toggleGameOfLifeCycle() {
+    resizeCanvasToDisplaySize(canvas);
+    if (stop) {
+        stop = false;
+        startGameOfLifeCycle();
+    } else {
+        stop = true;
+        ctx.clearRect(0, 0, canvas.width, canvas.height);
+    }
+}
+
+function startGameOfLifeCycle() {
+    seedGrid();
+    drawGrid();
+    
+    then = Date.now();
+    startTime = then;
+    gameOfLifeCycle();
+}
+
+function gameOfLifeCycle() {
+    if (stop) {
+        return;
+    }
+    
+    requestAnimationFrame(gameOfLifeCycle);
+    now = Date.now();
```

Comment:
Is it true that you only need the "now" variable inside the gameOfLifeCycle() function? If so, I would declare it within this function rather than globally. As a generally rule, you should aim for your variables to local rather than global whenever possible.